indico/modules/events/registration/client/js/form/fields/TimetableSessions.jsx

```diff
@@ -0,0 +1,323 @@
+  onFocus: PropTypes.func.isRequired,
+  onBlur: PropTypes.func.isRequired,
+  allowFullDays: PropTypes.bool.isRequired,
+  display: PropTypes.oneOf(['expanded', 'collapsed']),
+  maximum: PropTypes.number.isRequired,
+  minimum: PropTypes.number.isRequired,
+};
+
+TimetableSessionsComponent.defaultProps = {
+  sessionData: {},
+  display: 'expanded',
+};
+
+function SessionBlockHeader({
+  value,
+  index,
+  data,
+  label,
+  expandedHeaders,
+  allowFullDays,
+  handleValueChange,
+  onChange,
+  onClick,
+}) {
+  const isExpanded = expandedHeaders.includes(index);
+  const childrenIds = data.map(e => e.id);
+  const isAllChecked = childrenIds.every(id => value.includes(id));
+  const isAnyChecked = childrenIds.some(id => value.includes(id));
+  const selectedBlocks = value.filter(element => childrenIds.includes(element)).length;
+
+  function handleHeaderChange(e) {
+    e.stopPropagation();
+    const newValue =
+      isAllChecked || isAnyChecked
+        ? value.filter(element => !childrenIds.includes(element))
+        : [...value, ...childrenIds];
+    onChange(newValue);
+  }
+
+  return (
+    <Accordion styled exclusive={false}>
+      <Accordion.Title active={isExpanded} index={index} onClick={onClick}>
+        <Icon name="dropdown" />
+        <Checkbox
+          checked={isAllChecked}
+          onChange={handleHeaderChange}
+          indeterminate={isAnyChecked && !isAllChecked}
+          disabled={!allowFullDays}
+          label={label}
+        />
+        <span>
+          <Translate>{`(${selectedBlocks} blocks selected)`}</Translate>
+        </span>
+      </Accordion.Title>
+      <Accordion.Content active={isExpanded}>
+        {data.map(({fullTitle, id}) => (
+          <dd className="grouped-fields" key={id}>
+            <Checkbox
+              value={id}
+              onChange={handleValueChange}
+              checked={value.includes(id)}
+              label={fullTitle}
+            />
+          </dd>
+        ))}
+      </Accordion.Content>
+    </Accordion>
+  );
+}
+
+SessionBlockHeader.propTypes = {
+  value: PropTypes.arrayOf(PropTypes.number).isRequired,
+  index: PropTypes.number.isRequired,
+  data: PropTypes.arrayOf(
+    PropTypes.shape({
+      full_title: PropTypes.string,
+      id: PropTypes.number,
+      start_dt: PropTypes.string,
+    })
+  ).isRequired,
+  label: PropTypes.string.isRequired,
+  expandedHeaders: PropTypes.arrayOf(PropTypes.number).isRequired,
+  allowFullDays: PropTypes.bool.isRequired,
+  handleValueChange: PropTypes.func.isRequired,
+  onChange: PropTypes.func.isRequired,
+  onClick: PropTypes.func.isRequired,
+};
+
+export default function TimetableSessionsInput({
+  htmlId,
+  htmlName,
+  disabled,
+  isRequired,
+  allowFullDays,
+  display,
+  minimum,
+  maximum,
+}) {
+  const {eventId} = useSelector(getStaticData);
+  const {data: sessionData} = camelizeKeys(
+    useIndicoAxios({url: timeTableURL({event_id: eventId})})
+  );
```

Comment:
I found out you can actually pass `camelize: true` directly:

```suggestion
  const {data: sessionData} = useIndicoAxios({url: timeTableURL({event_id: eventId})}, {camelize: true});
```